### PR TITLE
Doc Js Driver: Basic usage examples

### PR DESCRIPTION
**Pull request by michielmulders**

### CODE REVIEW

**Comment by kremalicious** [comment]
hey @michielmulders, that's cool, thanks! But the file permission changes need to be somehow removed from this PR. You changed the permission of each and every file in this repo from `644` to `755`. 

Maybe revert via git locally, add your changes back on top again and force push to this branch. Or just change all permissions back and commit that change, which should remove it from this PR too. Let me know if you need help with that, git can get tricky with accidental published changes 🥃

File: docs/source/usage.rst [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+Basic Usage Examples
+====================
+
+For the examples on this page, we assume you've :doc:`installed the bigchaindb_driver JavaScript package <quickstart>`,
+and you have determined the BigchainDB Root URL (issue: move this to general docs)
+of the node or cluster you want to connect to.
+
+This example guides you through creating and transferring an asset.
+We walk through the code explaining its use, some pieces are left out
+because they have no real use (e.g. defenition of global variable)
```
Comment (TimDaub) [inline docs/source/usage.rst]: `definition`

File: docs/source/usage.rst [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+*Full working code* can be found at the bottom of this document.
+The following code are just snippets.
+
+Getting Started
+---------------
+We begin by creating an object of BigchainDB driver:
```
Comment (TimDaub) [inline docs/source/usage.rst]: `We begin by importing the BigchainDB driver`

File: docs/source/usage.rst [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+Next, we define a constant containing the API path.
+
+.. code-block:: js
+
+	const API_PATH = 'http://localhost:9984/api/v1/'
+
+Create Connection With BigchainDB
+---------------------------------
+
+A simpel connection with BigchainDB can be established like this.
```
Comment (TimDaub) [inline docs/source/usage.rst]: `simple`

File: docs/source/usage.rst [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+
+	const conn = new driver.Connection(API_PATH)
+
+It is also possible to connect to a BigchainDB node of the IPDB test net.
+To do so, you need to pass the **app_id and app_key**. 
+
+.. code-block:: js 
+
+	let bdb = new driver.Connection('https://test.ipdb.io/api/v1/', { 
+		app_id: 'dgi829l9',
```
Comment (TimDaub) [inline docs/source/usage.rst]: I hope these are not your actual keys to use the test net. If so please remove them from this PR.

File: docs/source/usage.rst [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+transaction:
+
+.. code-block:: js
+	
+	const txCreateAliceSimple = driver.Transaction.makeCreateTransaction(
+   		assetdata,
+   		metadata,
+
+		// A transaction needs an output
+		[ driver.Transaction.makeOutput(
```
Comment (TimDaub) [inline docs/source/usage.rst]: I think you should explain between l104 - l107 what inputs and outputs in a BigchainDB transaction are.

File: docs/source/usage.rst [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+    confirms a transaction as being valid.
+
+Asset Transfer
+--------------
+
+Imagine some time goes by, during which Alice is happy with her bicycle, and
+one day, she meets Bob, who is interested in acquiring her bicycle. The timing
+is good for Alice as she had been wanting to get a new bicycle.
+
+To transfer the bicycle (asset) to Bob, Alice must consume the transaction in
```
Comment (TimDaub) [inline docs/source/usage.rst]: `Alice must consume the transaction's output`

File: docs/source/usage.rst [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+which the Bicycle asset was created.
+
+Alice could retrieve the transaction:
+
+.. code-block:: js
+
+	driver.Connection.getTransaction(txCreateAliceSimpleSigned.id)
```
Comment (TimDaub) [inline docs/source/usage.rst]: `conn.getTransaction`

File: docs/source/usage.rst [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+
+.. code-block:: js
+
+	txTransferBobSigned = driver.Transaction.signTransaction(txTransferBob, alice.privateKey);
+
+And sent over to a BigchainDB node:
+
+.. code-block:: js
+
+	conn.postTransaction(txTransferBobSigned, API_PATH)
```
Comment (TimDaub) [inline docs/source/usage.rst]: Remove `API_PATH`

File: docs/source/usage.rst [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+
+.. code-block:: js
+
+	conn.pollStatusAndFetchTransaction(txTransferBobSigned.id)
+
+Bob is the new owner:
+
+.. code-block:: js
+
+	console.log('Is Bob the owner?', tx['outputs'][0]['public_keys'][0] == bob.publicKey)
```
Comment (TimDaub) [inline docs/source/usage.rst]: `tx`is not defined anywhere.

File: docs/source/usage.rst [hunk 10]
```diff
@@ -0,0 +1,364 @@
...
+	   {'data': {'bicycle': {'serial_number': 'abc', manufacturer: 'Bicycle Inc.'}}},
+	   {'data': {'bicycle': {'serial_number': 'cde', manufacturer: 'Bicycle Inc.'}}},
+	   {'data': {'bicycle': {'serial_number': 'fgh', manufacturer: 'Bicycle Inc.'}}}
+	]
+
+Let’s perform a text search for all assets that contain the word 'Bicycle Inc.':
+
+.. code-block:: js
+
+	let assets = conn.searchAssets('Bicycle Inc.')
```
Comment (TimDaub) [inline docs/source/usage.rst]: `let assets` might never really get defined, as `searchAssets` returns a promise.

File: docs/source/usage.rst [hunk 11]
```diff
@@ -0,0 +1,364 @@
...
+			const txTransferBob = driver.Transaction.makeTransferTransaction(
+				// signedTx to transfer
+				txCreateAliceSimpleSigned,
+				// metadata
+				{price: '100 euro'},
+				[driver.Transaction.makeOutput(driver.Transaction.makeEd25519Condition(bob.publicKey))], 
+				0)
+			
+			// Sign with alice's private key
+			txTransferBobSigned = driver.Transaction.signTransaction(txTransferBob, alice.privateKey)
```
Comment (TimDaub) [inline docs/source/usage.rst]: Needs `let`?

File: docs/source/usage.rst (hunk 4 above)
Comment (michielmulders) [reply docs/source/usage.rst]: No, fake keys! :)

Comment (michielmulders) [reply docs/source/usage.rst]: But correct lenght

File: docs/source/usage.rst (hunk 9 above)
Comment (michielmulders) [reply docs/source/usage.rst]: tx is defined in the code at bottom of document, this is just a snippet?
Would you do it differently?

File: docs/source/usage.rst (hunk 10 above)
Comment (michielmulders) [reply docs/source/usage.rst]: Idd, removed.

**Comment by michielmulders** [comment]
@kremalicious I changed the file permissions, can you confirm? @TimDaub can you do a review on the new commits. Changed the smaller stuff and added info about input/outputs (copied a bit from docs and source code).

File: docs/source/usage.rst (hunk 4 above)
Comment (TimDaub) [reply docs/source/usage.rst]: 👍

File: docs/source/usage.rst (hunk 11 above)
Comment (TimDaub) [reply docs/source/usage.rst]: This still isn't defined anywhere higher up, so needs a variable declaration? `let, const, var`?

**Comment by TimDaub** [comment]
Merged via #99.

Great work @michielmulders 🎉

**Pull request merged**